From b2ea88e70345d1d9cb2ebce095cc6f2e500618e9 Mon Sep 17 00:00:00 2001
From: Loren Huang <b02279@freescale.com>
Date: Wed, 24 Sep 2014 16:20:25 +0800
Subject: [PATCH 466/505] ENGR00332958 gpu: Add dumpgpu.sh to unit test

This is a script to get gpu information.
You can follow the pid or the application name to dump the gpu memory information for specific
running application.

Sample out on mx6dq:
GPU Info
gpu      : 0
model    : 2000
revision : 5108

gpu      : 1
model    :  320
revision : 5007

gpu      : 2
model    :  355
revision : 1215

VIDEO MEMORY:
    gcvPOOL_SYSTEM:
        Free  :          0 B
        Used  :          0 B
        Total :          0 B
    gcvPOOL_CONTIGUOUS:
        Used  :    4336128 B
    gcvPOOL_VIRTUAL:
        Used  :          0 B

NON PAGED MEMORY:
    Used  :          0 B
Paged memory Info
low:  1695744 bytes
high: 0 bytes
CMA memory info
cma:  4341760 bytes
VidMem Usage (Process 1911):
                All     Index    Vertex   Texture        RT     Depth    Bitmap        TS     Image      Mask   Scissor   HZDepth
Current     4336128         0         0         0         0         0   4336128         0         0         0         0         0
Maximum     5718528         0         0         0         0         0   5718528         0         0         0         0         0
Total     185117184         0         0         0         0         0 185117184         0         0         0         0         0
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Idle percentage:0.00%
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

Signed-off-by: Loren Huang <b02279@freescale.com>
---
 test/gpu/Makefile   |  3 ++-
 test/gpu/dumpgpu.sh | 37 +++++++++++++++++++++++++++++++++++++
 2 files changed, 39 insertions(+), 1 deletion(-)
 create mode 100755 test/gpu/dumpgpu.sh

diff --git a/test/gpu/Makefile b/test/gpu/Makefile
index 40c2eea..d8f50ea 100644
--- a/test/gpu/Makefile
+++ b/test/gpu/Makefile
@@ -2,7 +2,8 @@
 INCLUDE_LIST:=IMX6Q IMX6SL IMX6SX
 
 ifeq ($(PLATFORM),$(findstring $(PLATFORM),$(INCLUDE_LIST)))
-OBJS = $(OBJDIR)/gpu.sh
+OBJS = $(OBJDIR)/gpu.sh \
+	   $(OBJDIR)/dumpgpu.sh
 else
 OBJS =
 endif
diff --git a/test/gpu/dumpgpu.sh b/test/gpu/dumpgpu.sh
new file mode 100755
index 0000000..b99dee2
--- /dev/null
+++ b/test/gpu/dumpgpu.sh
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+input=$1
+echo "GPU Info"
+cat /sys/kernel/debug/gc/info
+cat /sys/kernel/debug/gc/meminfo
+echo "Paged memory Info"
+cat /sys/kernel/debug/gc/allocators/default/lowHighUsage
+echo "CMA memory info"
+cat /sys/kernel/debug/gc/allocators/cma/cmausage
+#echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+time=`cat /sys/kernel/debug/gc/idle`
+#echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+if [ x$input != x ]
+then
+pidlist=`cat /sys/kernel/debug/gc/clients | awk '{if ($1 == '$input') print $1}'`
+if [ x$pidlist == x ]
+then
+pidlist=`cat /sys/kernel/debug/gc/clients | awk '{if ($2~/'$input'/) print $1}'`
+fi
+if [ x$pidlist == x ]
+then
+echo "Invalid input: $input"
+fi
+else
+pidlist=`cat /sys/kernel/debug/gc/clients | awk '{if ($1~/^[0-9]/) print $1}'`
+fi
+for i in $pidlist
+do
+    echo $i > /sys/kernel/debug/gc/vidmem
+    cat /sys/kernel/debug/gc/vidmem
+done
+echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+sleep 1
+time=`cat /sys/kernel/debug/gc/idle | awk '{printf "%.2f", $7/10000000.0}'`
+echo "Idle percentage:"$time"%"
+echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
-- 
1.8.0

